### ISSUE TITLE
pre builds broken by scipy 1.6.0rc1 

### ISSUE THREAD

**Issue by jni** [scikit-image/scikit-image#5130]
## Description

The release of scipy 1.6.0rc1 has broken our builds, see e.g. [here](https://github.com/scikit-image/scikit-image/pull/5129/checks?check_run_id=1540462654#step:5:442). The failing test is `skimage/segmentation/tests/test_boundaries.py::test_mark_boundaries_subpixel`. As I recall that function depends on a rather precise interpretation of the sizes given to `ndi.zoom`, so I expect that the culprit lies in recent (positive!) changes by @grlee77 to how SciPy handles interpolation. Probably a +1 can be removed somewhere and we can move on! 🤞

**Comment by grlee77** [comment]
It looks like inside `mark_boundaries` there is a call to `ndi.zoom` with `mode='reflect'`. This mode has had a long-standing bug in scikit-image that was finally resolved with 1.6 in scipy/scipy#12767.  In short the behavior of reflect was fixed from behaving like the top left panel in https://github.com/scipy/scipy/issues/2640#issuecomment-554873759 to the expected behavior in the top middle panel.

Do you know how the expected result for this test was originally generated? If it relied on this erroneous behavior then we probably just need to regenerate the reference. I will take a closer look in a bit, but if it is only this boundary change then the central part of the image should still match the refernce, but a couple pixel width at the border could be different.

**Comment by grlee77** [comment]
The simplest solution is probably to just change from `'reflect'` to `'mirror'` at this line:
https://github.com/scikit-image/scikit-image/blob/448ae9720645a4fd2ec78c9ceffa64d0905096a2/skimage/segmentation/boundaries.py#L223-L224

I verified that this will pass with the existing test data. There is actually a separate issue from the one above in that zoom never requested coordinate locations outside the original image bounds. However, the interpolation kernel does extend past the image boundary. For this case, in SciPy, the boundary mode was previously [ALWAYS hardcoded to mirror](https://github.com/scipy/scipy/pull/12767/commits/b232551d80f395a6ebd125d589e2495e9b58382a) when the center pixel of the interpolation kernel is within the image, but some portions of the filter kernel extended outside the boundary! In a subsequent commit, [this was fixed](https://github.com/scipy/scipy/pull/12767/commits/621515e6f848673e8453279a8fdb6a2a74f4a71c#diff-a9f2afb91465eaedc461b52f7a81911c5df511571b072dd50df95ccc7815e3f3R455), so we either have to update the reference result or switch the function to use mode='mirror' to get a result identical to the existing reference.

Let me know which you think is preferred. I don't see a strong reason to use `'mirror'` instead of `reflect` here, so maybe best to just switch it internally so that user results remain consistent across scikit-image versions.
